### DISCUSSION TITLE
How to convert custom FFMPEG command to go2RTC strean definition

### DISCUSSION THREAD

**Discussion by djtremolo** [blakeblackshear/frigate#6033] [Ask A Question] [answered]
Hi all!

In my frigate setup, I'm using go2RTC for restreaming main&sub streams. That works perfectly.

However, I'd like to generate the sub streams by downscaling the main streams (2560x1920@30fps -> 1280x960@5fps). I was able to do that using the software transcoding, but the CPU usage is unnecessarily high.

Without go2RTC, I can do what I want with the ffmpeg command below. But my question is: how do I get this rewritten to a go2RTC stream definition? Tried for quite many hours with no success :(

This works on command line:
```
ffmpeg -y -hide_banner 
  -hwaccel cuvid
  -hwaccel_output_format cuda
  -resize 1280x960 
  -c:v h264_cuvid 
  -i "rtmp://192.168.xx.xx/bcs/channel0_main.bcs?channel=0&stream=1&user=xxxxx&password=yyyy"
  -filter:v "fps=5" 
  -c:v h264_nvenc 
  -c:a copy 
  -preset slow 
  myOutputFilename.mp4
```

**Comment by NickM-27** [comment, marked as answer]
This is what I have on one of mine:

`  "ffmpeg:backyard_cam#video=h264#width=1280#height=960#hardware=cuda"`

**Reply by djtremolo** [reply]
Awesome, it works! Thank you!

My next goal is to limit the FPS. Apparently `#raw=-vf "fps=fps=5"` does not work - perhaps because it's for a software filter and now I'm using GPU encoding? Any NVIDIA friendly magic words?

Edit: hmm, the vf trick worked earlier when I ran the command on bash. Strange.

**Reply by djtremolo** [reply]
oh, got it pretty much working with this:
`mystream: ffmpeg:mysource#video=h264#width=1280#height=960#raw=-filter:v "fps=5"#hardware=cuda`

Now my VLC tells me FPS=8, but that's better than the original 30.

**Reply by NickM-27** [reply]
I have the same logic and this is what I use `#raw=-r 10` which sets it exactly
